### PR TITLE
Fix DeepSpeed warning CI Test

### LINKED ISSUE #9193: fix deepspeed special test `test_deepspeed_stage_3_save_warning`
### To Reproduce

Run on current master branch the following command:

`SPECIAL_PATTERN=test_deepspeed_stage_3_save_warning bash tests/special_tests.sh
`

Error:

```bash

______________________________ test_deepspeed_stage_3_save_warning _______________________________

tmpdir = local('/tmp/pytest-of-adrian/pytest-1376/test_deepspeed_stage_3_save_wa0')

    @RunIf(min_gpus=2, deepspeed=True, special=True)
    def test_deepspeed_stage_3_save_warning(tmpdir):
        """
        Test to ensure that DeepSpeed Stage 3 gives a warning when saving.
        """
        model = BoringModel()
        trainer = Trainer(
            default_root_dir=tmpdir, plugins=[DeepSpeedPlugin(stage=3)], gpus=2, fast_dev_run=True, precision=16
        )
        trainer.fit(model)
        checkpoint_path = os.path.join(tmpdir, "model.pt")
        with pytest.warns(UserWarning, match="each worker will save a shard of the checkpoint within a directory."):
>           trainer.save_checkpoint(checkpoint_path)
E           Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) was emitted. The list of emitted warnings is: [].

tests/plugins/test_deepspeed_plugin.py:432: Failed


```

### Environment

* CUDA:
        - GPU:
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
                - GeForce RTX 3090
        - available:         True
        - version:           11.1
* Packages:
        - numpy:             1.20.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.9.0+cu111
        - pytorch-lightning: 1.5.0dev
        - tqdm:              4.60.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.8
        - version:           #64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020

### Additional context

In our GPU CI as well, check section for "special tests"

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Fixes #9193.

Only check on rank 0 if the warning is thrown!

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
Screenshot 2021-09-03 at 16 40 43

LGTM

**Review by ethanwharris — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
🎉 thanks
